### ISSUE TITLE
otherModesKeyBindings does not work in Visual mode

### ISSUE THREAD

**Issue by DarylPinto** [VSCodeVim/Vim#2737]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bug Report

**What happened**:
Remapping a key in `vim.otherModesKeyBindings` works in Command mode but not Visual mode

**What did you expect to happen**:
I expected the key to behave the same in both modes.

**How to reproduce it (as minimally and precisely as possible)**:
```
"vim.otherModesKeyBindings":  [
        {
            "before": ["s"],
            "after": ["<Nop>"]
        }
]
```

You'll notice that the key is unbound in Command mode, but still active in Visual mode.

**Environment**:

- Extension (VsCodeVim) version: 0.12.0
- VSCode version: 1.24.1
- OS version: OSX 10.12.6

**Comment by ohjames** [comment]
Since VsCodeVim version 0.13.0 you have to use the `visual...` and `normal...` options instead of the `other...` and the `visual...` ones seem to not work at all.

**Comment by ohjames** [comment]
Think this should be closed in favor of #2233

**Comment by dylanscott** [comment]
It looks like `otherModeKeybindings` was renamed `normalModeKeybindings` in [this commit](https://github.com/VSCodeVim/Vim/commit/2a3d289760b4483fcf87ae65a341c151d7282d94).

This is the second time I've had my keybindings broken by an update and required me to read look through the commit history for this extension to fix them. It's very frustrating. I don't understand why you wouldn't make a change like this in a backwards compatible way that wouldn't break people currently using `otherModeKeybindings`.

**Comment by jpoon** [comment]
@dylanscott this was mentioned in our changelog -- https://github.com/VSCodeVim/Vim/blob/master/CHANGELOG.md#v0130-2018-06-18. I also put out a notice on our Slack that this was coming.

**Comment by yellowiscool** [comment]
In my use case, I replaced my `otherModesKeyBindings` array by two identical `normalModeKeyBindings` and `visualModeKeyBindings` arrays.

A suggestion for breaking changes could be a small warning appearing one version before the breaking change.

**Comment by DarylPinto** [comment]
@jpoon The issue I'm having still persists in 0.13.0. However, upon further investigation it appears that the cause of the issue is slightly different than I initially thought.

When using Vintage Mode in Sublime Text or Vi in the MacOS Terminal, pressing `v` to enter Visual Mode shows `-- VISUAL --` in the statusline. However, using VSCodeVim the statusline reads `-- VISUAL -- v`, implying that the `v` key was pressed again after entering visual mode. This appears to be the cause of the problem, as moving the caret back and forth such that the statusline no longer says `v` fixes the problem.

**Comment by ohjames** [comment]
Moving the caret back and forth fixes it due to the reasons discussed in #2233.
